model/jpa/src/main/java/org/keycloak/storage/jpa/entity/FederatedUserAttributeEntity.java

```diff
@@ -79,11 +81,23 @@ public void setName(String name) {
     }
 
     public String getValue() {
-        return value;
+        if (value != null && long_value != null) {
+            throw new IllegalStateException(String.format("Federated user with id %s should not have set both `value` and `long_value`.", userId));
```

Comment:
Same thing here regarding the attribute name - if something goes wrong, the name helps to narrow down the faulty attribute.